dotlottie-rs/src/dotlottie_player.rs

```diff
@@ -1141,6 +1184,598 @@ impl DotLottiePlayer {
+                                }
+                            }
+                            Err(_) => return vec![],
+                        }
+                    }
+                    listener_types
+                } else {
+                    vec![]
+                }
+            }
+            Err(_) => vec![],
+        }
+    }
+
+    // todo: Once lister actions are implemented, remove this fn
+    pub fn tmp_set_state_machine_context(&self, key: &str, value: f32) -> bool {
+        if self.state_machine.read().unwrap().is_none() {
+            return false;
+        }
+
+        self.state_machine
+            .write()
+            .unwrap()
+            .as_mut()
+            .unwrap()
+            .set_numeric_context(key, value);
+
+        true
+    }
+
+    pub fn tmp_set_state_machine_string_context(&self, key: &str, value: &str) -> bool {
+        if self.state_machine.read().unwrap().is_none() {
+            return false;
+        }
+
+        self.state_machine
+            .write()
+            .unwrap()
+            .as_mut()
+            .unwrap()
+            .set_string_context(key, value);
+
+        true
+    }
+
+    pub fn tmp_set_state_machine_bool_context(&self, key: &str, value: bool) -> bool {
+        if self.state_machine.read().unwrap().is_none() {
+            return false;
+        }
+
+        self.state_machine
+            .write()
+            .unwrap()
+            .as_mut()
+            .unwrap()
+            .set_bool_context(key, value);
+
+        true
+    }
+
+    pub fn post_event(&self, event: &Event) -> bool {
+        if self.state_machine.read().unwrap().is_none() {
+            return false;
+        }
+
+        match self.state_machine.try_write() {
+            Ok(mut state_machine) => {
+                if let Some(sm) = state_machine.as_mut() {
+                    sm.post_event(event);
+                }
+            }
+            Err(_) => return false,
+        }
+
+        true
+    }
+
+    /// Post event format:
+    ///
+    /// "Bool: true"
+    /// "Bool: false"
+    /// "String: ..."
+    /// "Numeric: 0.0"
+    /// "OnPointerDown: 0.0 0.0"
+    /// "OnPointerUp: 0.0 0.0"
+    /// "OnPointerMove: 0.0 0.0"
+    /// "OnPointerEnter: 0.0 0.0"
+    /// "OnPointerExit: 0.0 0.0"
+    /// "OnComplete"
+    #[cfg(target_arch = "wasm32")]
+    pub fn post_event_payload(&self, event: String) -> bool {
+        if self.state_machine.read().unwrap().is_none() {
+            return false;
+        }
+
+        let parts: Vec<&str> = event.splitn(2, ": ").collect();
+        if parts.len() < 2 {
+            return false;
+        }
+
+        let command_type = parts[0];
+        let value = parts[1];
+
+        match command_type {
+            "Bool" => {
+                let bool_value = value.parse::<bool>().map_err(|_| ());
+                let bool_event = Event::Bool {
+                    value: bool_value.unwrap(),
```

Comment:
instead of unwrap, we may return false instead

Reply:
please handle this in other instances as well in the same function

Reply:
what do you mean? we need to unwrap to get the value

Reply:
@samuelOsborne alright, wouldn't unwrap return a Result ? so on the Err part we should return false, correct me if i'm wrong

Reply:
@theashraf I see what you mean now, have added a safer way to get the parsed value